lib/finch/http1/pool.ex

```diff
@@ -55,7 +55,23 @@ defmodule Finch.HTTP1.Pool do
     catch
       :exit, data ->
         Telemetry.exception(:queue, start_time, :exit, data, __STACKTRACE__, metadata)
-        exit(data)
+
+        # Provide helpful error messages for known errors
+        case data do
+          {:timeout, {NimblePool, :checkout, pids}} ->
+            reraise(
+              "
+            Timeout while waiting to checkout an http connection for process(es): #{inspect(pids)}.
+            Consider:
+            1. Increasing your poolsize (recommended).
```

Comment:
I really like the intentionality behind this error message, but I don't know that we should make either of these recommendations. If the user is receiving this error, the obvious implication is that the connections are all being utilized longer than the request is willing to wait. This means we have a queue building up which further implies that our arrival rate is exceeding the work time of the downstream service.

Increasing the pool timeout is just going to allow that queue to grow even deeper making the problem worse by further taxing the systems resources.
Increasing the connection count per pool or adding a new pool _may_ help alleviate the issue. But, if the downstream system is becoming overloaded then adding additional load results in more queueing in the users service further degrading performance.

In any case, I don't feel like we should make either of these recommendations and instead should try to be verbose about _why_ they're seeing timeouts. Maybe something like:

```
Finch was unable to provide a connection within the timeout due to excess queueing for connections. This could be because the downstream system can't keep up with the amount of requests being sent or because you have an insufficient number of connections and need to increase your pool counts.
```

Reply:
I'm all for it! I totally agree that the current message could lead a user down a poor path.

Any sort of language that doesn't lead a user to believe its a nimble pool problem at first glance is great IMO. As someone who has dealt with these cryptic-esque errors in hackney in production it can be hard to see the forest from the trees in the default error and understand exactly what you need to do to begin to ameliorate the issue.

I'll make the change to your verbiage.

Reply:
change has been made verbatim.

Reply:
> ```
> Finch was unable to provide a connection within the timeout due to excess queueing for connections. This could be because the downstream system can't keep up with the amount of requests being sent or because you have an insufficient number of connections and need to increase your pool counts.
> ```

In my experience the pool `count` should be increased if you see the pool becoming a bottleneck (message queue pile up) or also in some cases when you you see the [queue duration](https://github.com/keathley/finch/blob/main/lib/finch/telemetry.ex#L26) increase. In the case of checkout timeouts, you usually want to try increasing the `size` first.